src/main/scala/is/hail/expr/ir/functions/MathFunctions.scala

```diff
@@ -0,0 +1,86 @@
+package is.hail.expr.ir.functions
+
+import is.hail.expr.ir._
+import is.hail.expr.types._
+import is.hail.stats
+import org.apache.commons.math3.special.Gamma
+
+object MathFunctions extends RegistryFunctions {
+
+  def log(x: Double, b: Double): Double = math.log(x) / math.log(b)
+
+  def gamma(x: Double): Double = Gamma.gamma(x)
+
+  def rnorm(mean: Double, sd: Double): Double = mean + sd * scala.util.Random.nextGaussian()
+
+  def floor(x: Float): Float = math.floor(x).toFloat
+
+  def floor(x: Double): Double = math.floor(x)
+
+  def ceil(x: Float): Float = math.ceil(x).toFloat
+
+  def ceil(x: Double): Double = math.ceil(x)
+
+  def floorMod(x: Int, y: Int): Int = java.lang.Math.floorMod(x, y)
```

Comment:
can't these all be registered directly? e.g. `registerJavaStaticFunction[java.lang.Math]("%", tv("T", _.isInstanceOf[TIntegral]), tv("T"), tv("T"))("floorMod")`

Reply:
I guess some of these have to cast back to the original type...

Reply:
Yeah, definitely floorMod and floorDiv can be.  I'll change that.  For the ones that need the cast back, do you have a preference for consistency or should I convert one but not the other?

Reply:
I don't have a preference.

Reply:
OK, I left the mixed ones as is but converted floorMod.